### DISCUSSION TITLE
a patch function to extract patch's name

### DISCUSSION THREAD

**Discussion by ahmadtourei** [DASDAE/dascore#460] [General]
It would be nice to implement a function that can quickly report patch's name (without file extension) in directory spools. I currently use the following code to get patch's name:

```{python}
import dascore as dc

mem_sp = dc.examples.random_spool()
dir_sp = dc.examples.spool_to_directory(mem_sp)
dir_sp = dc.spool(dir_sp)

# extract the second patch's name
patch_name = os.path.basename(os.path.splitext(list(dir_sp.get_contents()["path"])[1])[0])
print(patch_name)
```

@d-chambers what do you think?

**Comment by d-chambers** [comment]
This might be a little cleaner:

```python
from pathlib import Path

Path(dir_sp.get_contents()['path'][0]).name
```

Or if you want all of them:

```python 
contents = dir_sp.get_contents()
name_series = contents['path'].str.split('.', expand=True)[0]
```

What would the API of the function look like?

**Comment by ahmadtourei** [comment]
Yep that's easier. But to avoid file extension, we need to use splitext:

`os.path.splitext(Path(dir_sp.get_contents()['path'][1]).name)[0]`

Could it be a utility patch function in [dascore.utils.patch](https://github.com/DASDAE/dascore/blob/v0.1.4/dascore/utils/patch.py#L1-L1088) with `file_format` argument? Something like `patch.name(file_format=False)`

**Comment by d-chambers** [comment]
closed by #471
